File: RegressionTests/Compiler/Sources/TestString.cs
using Junit.Framework;

namespace Dot42.Tests.Compiler.Sources
{
    public class TestString : TestCase
    {
        public void testLength()
        {
            var s = "Hello";
            AssertEquals(s.Length, 5);
        }

        public void testCharAt()
        {
            var s = "Hello";
            AssertEquals(s[0], 'H');
        }

        public void testIsNullOrEmpty1()
        {
            AssertEquals(string.IsNullOrEmpty(null), true);
        }

        public void testIsNullOrEmpty2()
        {
            AssertEquals(string.IsNullOrEmpty(""), true);
        }

        public void testIsNullOrEmpty3()
        {
            var s = "Hello";
            AssertEquals(string.IsNullOrEmpty(s), false);
        }

        public void testConcat1()
        {
            var s = "Hello";
            AssertEquals(s, "Hello");
            AssertEquals(s + "-there", "Hello-there");
        }

        public void testConcat2()
        {
            var s = "Hello";
            AssertEquals(s, "Hello");
            AssertEquals(s + 5, "Hello5");
        }

        public void testConcat3()
        {
            var s = "Hello";
            AssertEquals(s, "Hello");
            AssertEquals(s + 'a', "Helloa");
        }

        public void testEnumerable()
        {
            var s = "Hello";

            int i = 0;
            foreach (var c in s)
            {
                AssertEquals(s[i], c);
                i++;
            }
        }
    }
}

File: RegressionTests/JavaCompiler/JavaSources/TestString.java
package dot42.java.test;

import junit.framework.*;

public class TestString extends TestCase
{
	public void testLength()
	{
		String s = "Hello";
		assertEquals(s.length(), 5);
	}

	public void testCharAt()
	{
		String  s = "Hello";
		assertEquals(s.charAt(0), 'H');
	}

	public void testIsNullOrEmpty2()
	{
		assertEquals("".isEmpty(), true);
	}

	public void testIsNullOrEmpty3()
	{
		String  s = "Hello";
		assertEquals(s.isEmpty(), false);
	}

	public void testConcat1()
	{
		String s = "Hello";
		assertEquals(s, "Hello");
		assertEquals(s + "-there", "Hello-there");
	}

	public void testConcat2()
	{
		String s = "Hello";
		assertEquals(s, "Hello");
		assertEquals(s + 5, "Hello5");
	}

	public void testConcat3()
	{
		String s = "Hello";
		assertEquals(s, "Hello");
		assertEquals(s + 'a', "Helloa");
	}
}
